tempora/timing.py

```diff
@@ -220,45 +232,54 @@ class BackoffDelay(collections.abc.Iterator):
     True
     """
 
-    delay = 0
-
     factor = 1
     "Multiplier applied to delay"
 
-    jitter = 0
-    "Number or callable returning extra seconds to add to delay"
+    jitter: collections.abc.Callable[[], float]
+    "Callable returning extra seconds to add to delay"
 
     @jaraco.functools.save_method_args
-    def __init__(self, delay=0, factor=1, limit=float('inf'), jitter=0):
+    def __init__(
+        self,
+        delay: float = 0,
+        factor=1,
+        limit: collections.abc.Callable[[float], float] | float = float('inf'),
+        jitter: collections.abc.Callable[[], float] | float = 0,
+    ) -> None:
         self.delay = delay
         self.factor = factor
-        if isinstance(limit, numbers.Number):
+        if isinstance(limit, (numbers.Number, int, float)):
```

Comment:
I used `numbers.Number` because I wanted to include floats and ints and other numbers that are [rich comparable](https://github.com/python/typeshed/blob/f87041e1184db224520f038a05399d8cc3963f65/stdlib/builtins.pyi#L1523) to `int`. Is that `RichComparable[int]`? Same for the ` | float`s above.

Reply:
I found a cleaner solution where I can point to a possible mypy improvement when it comes to redefinitions: # python/mypy#16946 or # python/mypy#13914